### PR TITLE
Update google-apis page

### PR DESCRIPTION
**Pull request by sfshaza2**
Fixes https://github.com/flutter/website/issues/9313

The page need more than a link (to a page that doesn't even seem to exist).

@domesticmouse, @johnpryan, @craiglabenz, @ericwindmill, please review! I'm also interested in other resources that might be available now that weren't when the page was written. For example, are there more resources/examples for cloud authentication or other types of authentication?

### CODE REVIEW

File: src/data-and-backend/google-apis.md [hunk 1]
```diff
@@ -5,34 +5,43 @@ description: How to use Google APIs with Flutter.
...
-
+  The only APIs you should use directly from your Flutter
+  project are those that access user data using Google authentication.
+
+  APIs that require [service accounts][] **should not**
+  be used directly from a Flutter application.
+  Doing so requires shipping service credentials as part
+  of your application, which is not secure.
+  To use these APIs,
+  we recommend creating an intermediate service.
```
Comment (johnpryan) [inline src/data-and-backend/google-apis.md:26]: Do we have an example of how to do this?

Comment (domesticmouse) [reply src/data-and-backend/google-apis.md:26]: Sounds like a good cookbook article idea

**Comment by domesticmouse** [comment]
Looks like the CI pipeline is having issues.

```console
Checking for valid link references...
check_link_references: Invalid link references found!

data-and-backend/google-apis/index.html:
[<code class="language-plaintext highlighter-rouge">YouTubeApi</code>][]
```

I'd love to see a staged version...

File: src/data-and-backend/google-apis.md (hunk 1 above)
Comment (sfshaza2) [reply src/data-and-backend/google-apis.md:26]: idk

**Review by johnpryan — APPROVED** [review]

File: src/data-and-backend/google-apis.md (hunk 1 above)
Comment (parlough) [reply src/data-and-backend/google-apis.md:26]: I'm also not aware of a sample we have, but a simple sample or cookbook recipe sounds great (https://github.com/flutter/website/issues/9755).

[Cocoon](https://github.com/flutter/cocoon) might use service accounts on its app engine backend, but I am not familiar with it and it's likely too complicated to be a useful example.

**Review by parlough — APPROVED** [review]
Thanks for this cleanup!

A few small suggestions and a fix :)

**Review by domesticmouse — APPROVED** [review]
LGTM once CI passes

**Pull request merged**